src/query/functions/src/scalars/variant.rs

```diff
@@ -228,6 +230,50 @@ pub fn register(registry: &mut FunctionRegistry) {
         }),
     );
 
+    registry
+        .register_combine_nullable_2_arg::<VariantType, ArrayType<StringType>, VariantType, _, _>(
+            "get_by_keypath",
```

Comment:
We can change this function to a factory function, so that the key paths only need to be parsed once, and we don't need to repeat the parser every time we execute it. See the `json_path_query` and `json_array` implementations.